### ISSUE TITLE
bug with firefox hover tool coordinates

### ISSUE THREAD

**Issue by chupach** [bokeh/bokeh#5986]
Here the sample code demonstrating a bug in Firefox hover tool coordinates.
(various late versions in osx) with various late versions of bokeh)
No faulty behaviour in Safari.

This happens when the browser window is smaller than the window itself (i.e. the window can be scrolled). In this case hover tool coordinates are wrong: 
-coordinates are shifted by the amount of scrolling

The attached image shows an example: the mouse location is represented by the star. Hover tooltip shows up at wrong coordinates, as can also be seen in the js console.

![firefox_coo_bug](https://cloud.githubusercontent.com/assets/20745855/23847581/5d408dca-07d3-11e7-8c57-20d49c7144ea.jpg)

```
from bokeh.plotting import figure
from bokeh.io import curdoc
from bokeh.models import ColumnDataSource,HoverTool,CustomJS

hover_tool = HoverTool()
hover_tool_callback = CustomJS(code="""console.log('x=',cb_data.geometry['x'],'y=',cb_data.geometry['y']);""")
hover_tool.callback = hover_tool_callback

p = figure(tools = [hover_tool])
cds = ColumnDataSource({'x':[1,2,3,4],'y':[1,2,3,4]})
p.line(x='x',y='y', source=cds)
curdoc().add_root℗
```

**Comment by chupach** [comment]
This bug prevents the hover tool of being of any use with Firefox, when the browser window is scrolled.
It sounds quite serious and a fix or a workaround would be more than welcome.
Thx !

**Comment by bryevdv** [comment]
@chupach Thanks for the report. No one wants bugs but the core team is extremely small. We will get to this as time and resources allow.

**Comment by bryevdv** [comment]
@chupach I can't repro this issue with the PR that was just merged, but please check out `0.12.5rc2` when it is availble to confirm it fixes for you

**Comment by chupach** [comment]
Hi Bryan,
yes. I confirm it fixes the firefox offset bug (0.12.5rc3). This is great.
Thanks for the magnificent software
